File: dojo_12/java/src/main/java/ClassRoom.java
import java.util.List;

public class ClassRoom implements DataBag {
    private final Capacity capacity;
    private final SquareMeter squareMeter;
    private final Computers computers;

    public ClassRoom(Label lab_a, Capacity capacity) {
        this.capacity = capacity;
        this.squareMeter = new SquareMeter();
        this.computers = new Computers();
    }

    public ClassRoom(Label label, Capacity capacity, SquareMeter squareMeter) {
        this.capacity = capacity;
        this.squareMeter = squareMeter;
        this.computers = new Computers();
    }

    public ClassRoom(Label lab_b, Capacity capacity, SquareMeter squareMeter, Computers computers) {
        this.capacity = capacity;
        this.squareMeter = squareMeter;
        this.computers = computers;
    }

    public boolean equals(Object other) {
        return other.getClass().equals(this.getClass());
    }

    public boolean fitsAll(List<Filter> filters) {
        boolean result = true;

        for( Filter f : filters) {
            if ( !f.fulfilledBy(this)) {
                result = false;
                break;
            }
        }

        return result;
    }

    @Override
    public Object value(String key) {
        switch (key) {
            case "squareMeter": return this.squareMeter;
            case "capacity": return this.capacity;
            case "computers": return this.computers;
        }
        return new None();
    }

    public LabelForBlind labelForBlind() {
        return new LabelForBlind();
    }
}

File: dojo_12/goland/class_room.go
package Room

type ClassRoom struct {
	properties map[string]interface{}
	label      *Label
}

func NewClassRoomFromMap(label *Label, properties map[string]interface{}) *ClassRoom {
	room := NewClassRoomLabeled(label)
	room.properties = properties
	return room
}

func NewClassRoom(label *Label, capacity *Capacity, area *Area) *ClassRoom {
	props := make(map[string]interface{})
	props["capacity"] = capacity
	props["area"] = area

	room := NewClassRoomFromMap(label, props)

	return room
}

func NewClassRoomLabeled(label *Label) *ClassRoom {
	room := new(ClassRoom)

	room.label = label

	return room
}

func (this ClassRoom) fitsAll(filters []Filter) bool {
	for _, f := range filters {
		if !f.fulfilledBy(this) {
			return false
		}
	}
	return true
}

func (this ClassRoom) value(propertyName string) interface{} {
	result, ok := this.properties[propertyName]

	if ok {
		return result
	}

	return nil

}

func (this *ClassRoom) getLabel() *Label {
	if this == nil {
		return NewLabel("No Room")
	}

	return this.label
}
